core/frontend/src/Viewport.ts

```diff
@@ -949,27 +949,29 @@ export abstract class Viewport implements IDisposable, TileUser {
 
     this.forEachMapTreeRef(async (tree) => promises.push(tree.getMapFeatureInfo(hit)));
 
-    const results = await Promise.all(promises);
+    const featureInfo: MapFeatureInfo = {};
 
     const worldPoint = hit.hitPoint.clone();
-    let cartoHitPoint: Cartographic|undefined;
     const backgroundMapGeometry = hit.viewport.displayStyle.getBackgroundMapGeometry();
     if (undefined !== backgroundMapGeometry) {
-      cartoHitPoint = await backgroundMapGeometry.dbToCartographicFromGcs(worldPoint);
+      featureInfo.hitPoint = await backgroundMapGeometry.dbToCartographicFromGcs(worldPoint);
     }
 
-    const featureInfo: MapFeatureInfo = {hitPoint: cartoHitPoint};
-    for (const result of results)
-      if (result !== undefined) {
+    try {
+      const results = await Promise.all(promises);
```

Comment:
If any one of the promises rejects, you'll get no results. Consider instead something like:
`this.forEachMapTreeRef(async (tree) => promises.push(tree.getMapFeatureInfo(hit)).catch(()) => undefined);`

Reply:
Please, do not resolve other people's comments on your PRs.